File: tests/test_data/Switch.java
package test_data;

class Switch {
    public static int simple() {
        int i = 1;

        switch (i) {
            case 1:
                return 2;
            case 200:
                return -1;
        }

        return 0;
    }

    public static int no_match() {
        int i = 1;

        switch (i) {
            case 2:
                return 2;
            case 200:
                return -1;
        }

        return 0;
    }

    public static int default_case() {
        int i = 5;
        i++;

        switch (i) {
            case 2:
                return 2;
            case 200:
                return -1;
            default:
                return 1;
        }
    }

    public static int table_switch() {
        int i = 0;

        switch (i) {
            case 1: return 1;
            case 2: return 2;
            case 3: return 3;
            default: return 4;
        }
    }
}

File: tests/switch.rs
use rjvm::vm::data_type::Value::Int;

#[path = "utils.rs"]
mod java;

#[test]
fn switch_simple() {
    assert_eq!(java::run_method("test_data/Switch", "simple"), Int(2));
}

#[test]
fn switch_no_match() {
    assert_eq!(java::run_method("test_data/Switch", "no_match"), Int(0));
}

#[test]
fn switch_default() {
    assert_eq!(java::run_method("test_data/Switch", "default_case"), Int(1));
}

#[test]
fn table_switch_simple() {
    assert_eq!(
        java::run_method("test_data/Switch", "table_switch_simple"),
        Int(2)
    );
}

#[test]
fn table_switch_default() {
    assert_eq!(java::run_method("test_data/Switch", "table_switch"), Int(4));
}
